td-shim/src/bin/td-shim/info.rs

```diff
@@ -0,0 +1,197 @@
+// Copyright (c) 2022 Intel Corporation
+//
+// SPDX-License-Identifier: BSD-2-Clause-Patent
+
+use alloc::vec::Vec;
+use core::mem::size_of;
+use core::str::FromStr;
+use log::error;
+use scroll::{Pread, Pwrite};
+use td_layout::build_time::{TD_SHIM_FIRMWARE_BASE, TD_SHIM_FIRMWARE_SIZE};
+use td_layout::memslice;
+use td_shim::metadata::*;
+use td_shim::speculation_barrier;
+use td_shim::{PayloadInfo, TdKernelInfoHobType, TD_ACPI_TABLE_HOB_GUID, TD_KERNEL_INFO_HOB_GUID};
+use td_uefi_pi::pi::guid::Guid;
+use td_uefi_pi::pi::hob::*;
+use td_uefi_pi::{fv, hob, pi};
+
+pub struct ShimInfo<'a> {
+    // If metadata contains the `TD_HOB` section, TD-Shim
+    // can get additional information from TD HOB
+    pub td_hob: Option<&'static [u8]>,
+
+    // If metadata contains one/more `PermMem` sections,
+    // TD-Shim should ignore the memory information in TD HOB.
+    metadata_has_perm: bool,
+
+    pub memory: Vec<ResourceDescription>,
+    pub acpi_tables: Vec<&'a [u8]>,
```

Comment:
I feel it is weird to put acpi_tables here.

I assume ShimInfo means the metadata / memory related static information, right?

Or you want to create a structure to hold Runtime information ?

Reply:
I agree that we should separate the different kind of informations. The `ShimInfo` is just derived from original `TdHobInfo` structure and combined `1` and `2`